lib/adapters/MochaAdapter.js

```diff
@@ -53,60 +52,64 @@ export default class MochaAdapter extends EventEmitter {
       parent = parent.parent
     }
 
-    var test = new Test(
-      testName,
-      suiteName
-    )
-    this.idDataMap[mochaTest.title] = test
-    return test
+    if (mochaTest.errors !== undefined) {
+      var status = (mochaTest.state === undefined) ? 'skipped' : mochaTest.state
+
+      return new Test(testName, suiteName, status, mochaTest.duration,
+          mochaTest.errors)
+    }
+
+    return new Test(testName, suiteName)
   }
 
   onStart () {
     var startSuite = this.convertSuite(this.runner.suite)
     startSuite.name = undefined
-    this.idDataMap.globalSuite = startSuite
 
     this.emit('runStart', startSuite)
   }
 
   onSuite (details) {
-    var suite = this.idDataMap[details.title]
-    if (suite !== this.idDataMap.globalSuite) {
-      this.emit('suiteStart', suite)
+    if (!details.root) {
+      this.emit('suiteStart', this.convertSuite(details))
     }
   }
 
   onTest (details) {
     this.errors = []
-    this.emit('testStart', this.idDataMap[details.title])
+    this.emit('testStart', this.convertTest(details))
   }
 
   onFail (test, error) {
     this.errors.push(error)
   }
 
   onTestEnd (details) {
-    var test = this.idDataMap[details.title]
+    var test
+
     if (details.pending) {
+      test = this.convertTest(details)
       this.emit('testStart', test)
-      test.status = 'skipped'
-    } else {
-      test.status = details.state
     }
-    test.runtime = details.duration
-    test.errors = this.errors
+
+    details.errors = this.errors
```

Comment:
We are adding it, because we need to stores the `errors` so that when we are emitting the suites on the suiteEnd event, the tests also contains error, status and runtime.